doc/manual/stacktraces.rst

```diff
@@ -0,0 +1,213 @@
+.. _man-stacktraces:
+
+.. currentmodule:: Base
+
+Stack Traces
+============
+
+The :mod:`StackTraces` module provides simple stack traces that are both human readable and easy to use programmatically.
+
+Viewing a stack trace
+---------------------
+
+The primary function used to obtain a stack trace is :func:`stacktrace`::
+
+    julia> using StackTraces
+
+    julia> stacktrace()
+    2-element Array{StackTraces.StackFrame,1}:
```

Comment:
I'm having a fair amount of trouble coming up with a way to make good doctests for most of these. Because the doctests aren't run from the REPL, every single stack trace generated by the doctests will be different from stack traces generated by the user. For example:

```
File "manual/stacktraces.rst", line 172, in default
Failed example:
    grandparent()
Expected:
    ERROR: Whoops!
    StackTrace with 5 StackFrames:
      child at none:1
      parent at none:1
      grandparent at none:3
      eval_user_input at REPL.jl:62
      [inlined code from REPL.jl:92] anonymous at task.jl:63
Got:
    ERROR: Whoops!
    StackTrace with 8 StackFrames:
      child at none:1
      parent at none:1
      grandparent at none:3
      eval at boot.jl:265
      [inlined code from sysimg.jl:14] eval_user_input at client.jl:96
      eval at boot.jl:265
      [inlined code from sysimg.jl:14] eval_user_input at client.jl:96
      _start at client.jl:338
```

I can change the stack trace examples in the documentation to match what will happen when `make -C doc doctest` is run, but I worry that this would make the documentation itself less useful to the user.